### PR TITLE
[ioctl] Build hdwallet delete command line into new ioctl

### LINKED ISSUE #3284: [ioctl] build hdwallet delete command line into new ioctl
### What would you like to be added
 - Step 1: create `hdwalletdelete.go` and `hdwalletdelete_test.go` files in `./ioctl/newcmd/hdwallet`. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletdelete.go in `hdwalletdelete.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletdelete.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description
Refactor `hdwalledelete` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3284 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewHdwalletCreateCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by LuckyPigeon** [comment]
This PR will continue after #3470 merged.

File: ioctl/newcmd/hdwallet/hdwalletdelete_test.go [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+
+package hdwallet
+
+import (
+	"testing"
+
+	"github.com/golang/mock/gomock"
+	"github.com/iotexproject/iotex-core/ioctl/config"
+	"github.com/iotexproject/iotex-core/ioctl/util"
+	"github.com/iotexproject/iotex-core/test/mock/mock_ioctlclient"
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletdelete_test.go:15]: iotex-core should be a single group

File: ioctl/newcmd/hdwallet/hdwalletdelete_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString", config.English).Times(4)
+
+	t.Run("delete hdwallet", func(t *testing.T) {
+		createClient := mock_ioctlclient.NewMockClient(ctrl)
+		password := "123"
+
+		createClient.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString", config.English).Times(9)
+		createClient.EXPECT().IsHdWalletConfigFileExist().Return(false).Times(3)
+		createClient.EXPECT().ReadSecret().Return(password, nil).Times(2)
+		createClient.EXPECT().WriteHdWalletConfigFile(gomock.Any(), gomock.Any()).Return(nil)
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletdelete_test.go:33]: no use in  `NewHdwalletDeleteCmd`

File: ioctl/newcmd/hdwallet/hdwalletdelete_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+		client.EXPECT().AskToConfirm(gomock.Any()).Return(true)
+		client.EXPECT().RemoveHdWalletConfigFile().Return(nil)
+
+		cmd := NewHdwalletCreateCmd(createClient)
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletdelete_test.go:38]: no need

**Review by huof6829 — APPROVED** [review]

**Review by Liuhaai — APPROVED** [review]

**Pull request merged**